Assets/Scripts/BehaviorTree/Node.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+
+namespace BehaviorTree
+{
+    public enum NodeState
+    {
+        RUNNING,
+        SUCCESS,
+        FAILURE
+    }
+
+    public class Node
+    {
+        protected NodeState State;
+
+        public Node Parent { get; private set; }
```

Comment:
We could set the property here and remove the default constructor.
```suggestion
        public Node Parent { get; private set; } = null;
```

Reply:
Cheh